components/textfield/index.css

```diff
@@ -501,6 +501,11 @@ governing permissions and limitations under the License.
 		&::placeholder {
 			color: var(--highcontrast-textfield-text-color-readonly, var(--mod-textfield-text-color-readonly, var(--spectrum-textfield-text-color-readonly)));
 			background-color: transparent;
+
+			.is-disabled &,
+			.is-disabled:hover & {
+				color: var(--highcontrast-textfield-text-color-disabled, var(--mod-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled)));
```

Comment:
Ensures we get a disabled color for placeholder text if both disabled and read-only are on